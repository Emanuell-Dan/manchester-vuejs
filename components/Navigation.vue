<template>
  <div class="nav">
    <div 
      class="nav__mobile md:hidden fixed z-20 text-center right-0 cursor-pointer mt-4 mr-4"
      @click="toggleMobileNav"
    >
      <span
        :class="{ 'nav__mobile-burger--open': isNavOpen }"
        class="nav__mobile-burger"
      />
      <span
        :class="{ 'nav__mobile-burger--open': isNavOpen }"
        class="nav__mobile-burger"
      />
      <span
        :class="{ 'nav__mobile-burger--open': isNavOpen }"
        class="nav__mobile-burger"
      />
      <span
        :class="{ 'nav__mobile-burger--open': isNavOpen }"
        class="nav__mobile-burger"
      />
    </div>
    <nav 
      :class="{'nav__desktop--show': isNavOpen}"
      class="nav__desktop fixed z-10 bg-black font-bold w-full"
    >
      <div class="nav__desktop-inner relative">
        <ul class="nav__list list-reset flex flex-col md:flex-row items-center md:justify-center mt-12 md:mt-4 mb-6 md:mb-4">
          <li 
            v-for="(item, index) in navItems"
            :key="index"
            class="nav__item cursor-pointer hover:text-white focus:text-white my-1 md:mx-2"
            @click="$emit('scrollToSection', item, isNavOpen = false)"
          >
            {{ item }}
          </li>
        </ul>
        <a
          href="https://github.com/Emanuell-Dan/manchester-vuejs"
          target="_blank"
          title="Manchester Vue.js GitHub repo"
          class="nav__link absolute h-5"
        />
      </div>
    </nav>
  </div>
</template>

<script>
const NAV_ITEMS = [
  'About',
  'Future events',
  'Past events',
  'News'
];

export default {
  name: 'Navigation',
  data() {
    return {
      navItems: NAV_ITEMS,
      isNavOpen: false
    }
  },
  methods: {
    toggleMobileNav() {
      this.isNavOpen = !this.isNavOpen;
    }
  }
};
</script>
