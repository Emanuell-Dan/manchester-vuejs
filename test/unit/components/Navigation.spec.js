import Navigation from '@/components/Navigation.vue';
import { mount } from '@vue/test-utils';

describe('Navigation.vue', () => {
	let cmp;

	beforeEach(() => {
		cmp = mount(Navigation);
	});

	it('should render with expected html structure', () => {
		expect(cmp.element).toMatchSnapshot();
	});

	describe('desktop', () => {
		it('nav is displayed ', () => {
			expect(cmp.find('.nav__desktop').exists()).toBeTruthy();
		});

		it('emits scrollToSection event when an item is clicked', () => {
			cmp.find('.nav__item:first-child').trigger('click');
			expect(cmp.emitted().scrollToSection).toBeTruthy();
			expect(cmp.vm.isNavOpen).toBeFalsy();
		});
	});


	describe('mobile', () => {
		window.innerWidth = 700; // < 768px

		it('mobile nav is displayed', () => {
			expect(cmp.find('.nav__mobile').exists()).toBeTruthy();
		});

		it('nav items are displayed open', () => {
			expect(cmp.find('.nav__mobile-burger').exists()).toBeTruthy();
		});

		it('nav items are closed', () => {
			cmp.find('.nav__mobile').trigger('click');
			expect(cmp.find('.nav__mobile-burger').classes()).toContain('nav__mobile-burger--open');
		});
	});
});
