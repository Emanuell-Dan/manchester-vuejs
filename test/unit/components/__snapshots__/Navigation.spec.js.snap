// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation.vue should render with expected html structure 1`] = `
<div
  class="nav"
>
  <div
    class="nav__mobile md:hidden fixed z-20 text-center right-0 cursor-pointer mt-4 mr-4"
  >
    <span
      class="nav__mobile-burger"
    />
     
    <span
      class="nav__mobile-burger"
    />
     
    <span
      class="nav__mobile-burger"
    />
     
    <span
      class="nav__mobile-burger"
    />
  </div>
   
  <nav
    class="nav__desktop fixed z-10 bg-black font-bold w-full"
  >
    <div
      class="nav__desktop-inner relative"
    >
      <ul
        class="nav__list list-reset flex flex-col md:flex-row items-center md:justify-center mt-12 md:mt-4 mb-6 md:mb-4"
      >
        <li
          class="nav__item cursor-pointer hover:text-white focus:text-white my-1 md:mx-2"
        >
          
          About
        
        </li>
        <li
          class="nav__item cursor-pointer hover:text-white focus:text-white my-1 md:mx-2"
        >
          
          Future events
        
        </li>
        <li
          class="nav__item cursor-pointer hover:text-white focus:text-white my-1 md:mx-2"
        >
          
          Past events
        
        </li>
        <li
          class="nav__item cursor-pointer hover:text-white focus:text-white my-1 md:mx-2"
        >
          
          News
        
        </li>
      </ul>
       
      <a
        class="nav__link absolute h-5"
        href="https://github.com/Emanuell-Dan/manchester-vuejs"
        target="_blank"
        title="Manchester Vue.js GitHub repo"
      />
    </div>
  </nav>
</div>
`;
